plantseg/viewer/widget/proofreading/split_merge_tools.py

```diff
@@ -0,0 +1,92 @@
+import numpy as np
+from plantseg.viewer.logging import napari_formatted_logging
+from skimage.segmentation import watershed
+
+from plantseg.viewer.widget.proofreading.utils import get_bboxes, get_idx_slice
+
+"""
+try:
+    import SimpleITK as sitk
+    from plantseg.segmentation.functional.segmentation import simple_itk_watershed_from_markers as watershed
+
+except ImportError:
+    from skimage.segmentation import watershed as skimage_ws
+    watershed = partial(skimage_ws, compactness=0.01)
+"""
+
+
+def _merge_from_seeds(segmentation, region_slice, region_bbox, bboxes, all_idx):
+    region_segmentation = segmentation[region_slice]
+
+    mask = [region_segmentation == idx for idx in all_idx]
+
+    new_label = 0 if 0 in all_idx else all_idx[0]
+
+    mask = np.logical_or.reduce(mask)
+    region_segmentation[mask] = new_label
+    bboxes[new_label] = region_bbox
+    napari_formatted_logging('Merge complete', thread='Proofreading tool')
+    return region_segmentation, region_slice, bboxes
+
+
+def _split_from_seed(segmentation, sz, sx, sy, region_slice, all_idx, offsets, bboxes, image, seeds_values, max_label):
+    local_sz, local_sx, local_sy = sz - offsets[0], sx - offsets[1], sy - offsets[2]
+
+    region_image = image[region_slice]
+    region_segmentation = segmentation[region_slice]
+
+    region_seeds = np.zeros_like(region_segmentation)
+
+    seeds_values += max_label
+    region_seeds[local_sz, local_sx, local_sy] = seeds_values
+
+    mask = [region_segmentation == idx for idx in all_idx]
+    mask = np.logical_or.reduce(mask)
+
+    new_seg = watershed(region_image, region_seeds, mask=mask, compactness=0.001)
```

Comment:
It helps when there is no boundary evidence in the predictions. So if you have no signal and run WD from seeds, the location of the boundary is arbitrary. At the same time, compactness gives a distance penalty so that it would place the segmentation boundary in between the two scribbles. 

Unfortunately, the skimage implementation of WS is by far the slowest, but I have tried both vigra and sitk versions, and qualitatively skimage was the best.

Reply:
makes sense, thx for the clarification!